src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -2075,18 +2204,20 @@ METHOD(kernel_ipsec_t, add_system_enterprise, status_t,
 	struct system_enterprise_map_enterprise enterprise = {
 		.term_vlan_id = term_vlan_id,
 		/** 'term_private_addr' is set afterwards */
+		/* 'neigh_sub_interface' is set afterwards */
 	};
 
 	host_addr_to_picomsg_addr(term_private_addr, &enterprise.term_private_addr);
+	strcpy(enterprise.neigh_sub_interface, neigh_sub_interface);
```

Comment:
Never use `strcpy()` unless in very special occasions. Use `safe_strncpy()` instead.